### PR TITLE
liquidation: Disallow liquidating assets with zero maint asset weight

### PR DESCRIPTION
**Pull request by ckamm**
This removes the oracle dependency for tokens with zero maint asset weight that are not borrowable.

### CODE REVIEW

**Review by microwavedcola1 — APPROVED** [review]

File: bin/liquidator/src/liquidate.rs [hunk 1]
```diff
@@ -370,20 +370,16 @@ impl<'a> LiquidateHelper<'a> {
             .filter_map(|(ti, effective)| {
                 // check constraints for liquidatable assets, see also has_possible_spot_liquidations()
                 let tokens = ti.balance_spot.min(effective.spot_and_perp);
-                let is_valid_asset = tokens >= 1;
+                let is_valid_asset = tokens >= 1 && ti.maint_asset_weight.is_positive();
                 let quote_value = tokens * ti.prices.oracle;
-                // prefer to liquidate tokens with asset weight that have >$1 liquidatable
-                let is_preferred =
```
Comment (Lou-Kamades) [inline bin/liquidator/src/liquidate.rs:376]: we don't care about preferred assets anymore?

**Review by Lou-Kamades — APPROVED** [review]

File: bin/liquidator/src/liquidate.rs (hunk 1 above)
Comment (ckamm) [reply bin/liquidator/src/liquidate.rs:376]: Indeed, previously the liquidator preferred to liquidate against assets that had asset weight (and at least $1 in deposits). Now that tokens with asset weight are the only ones that can be liquidated, the logic for it is no longer needed.

File: programs/mango-v4/src/health/cache.rs
Comment (farnyser) [inline programs/mango-v4/src/health/cache.rs:1]: Shouldn't the new condition be added to `has_possible_spot_liquidations` ?

Comment (ckamm) [reply programs/mango-v4/src/health/cache.rs:1]: Could you explain? https://github.com/blockworks-foundation/mango-v4/pull/867/files#diff-110f42497491be964f83fc2e8ddceda72faac825ae293a9d26f5bdc8b6635312R937 changes `has_possible_spot_liquidations()` to add the ` && ti.maint_asset_weight.is_positive()` for the asset side.

**Pull request merged**